### PR TITLE
Reimplement dependency graph using Mermaid

### PR DESCRIPTION
**Pull request by mcmire**
## Explanation

Rely on GitHub's Mermaid integration to render the dependency graph instead of Graphviz, and update it to incorporate the packages that have been added recently. This prevents developers from needing to install Graphviz in order to regenerate the graph.

**[👉 See rendered version 👈](https://github.com/MetaMask/core/tree/update-dependency-graph#modules)**

## References

(none)

## Changelog

(none)

## Checklist

- [x] I've updated the test suite for new or updated code as appropriate
- [x] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [x] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,12 +28,81 @@ This repository houses the following packages:
 
 Or, in graph form [^fn1]:
 
-![Dependency graph](assets/dependency-graph.png)
```
Comment (Gudahtt) [inline README.md:31]: Looks like the old file is still there. Presumably we can delete that

Comment (mcmire) [reply README.md:31]: Yup, good call.

**Comment by mcmire** [comment]
@legobeat Hmm, what sort of overlap do you mean?

There are different requirements for the extension and this repo that we have to keep in mind. In this case we are interested in producing a graph of packages that this repo houses. Since each package corresponds to a workspace, we can use Yarn's workspaces feature to find all of them. In the extension, however, we are interested in files, so we have to use `madge` to find all of them.

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Comment by mcmire** [comment]
^ Rebased & squashed to resolve conflicts.

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Pull request merged**